### PR TITLE
Create a tutorial on how to deploy the Chrony charm

### PR DESCRIPTION
**Pull request by weiiwang01**
### Overview

Create a tutorial on how to deploy the Chrony charm

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Comment by weiiwang01** [comment]
@erinecon Could you please help to take a look at this chrony tutorial? Thanks!

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+* [Join our online chat](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)
+* [Contribute](https://charmhub.io/chrony/docs/contributing)
+* [Roadmap](https://charmhub.io/chrony/docs/roadmap)
+Thinking about using the Chrony charm for your next project? [Get in touch](https://matrix.to/#/#charmhub-charmdev:ubuntu.com)!
+
+## Contributing to this documentation
+
+Documentation is an important part of this project, and we take the same open-source approach to the documentation as the code. As such, we welcome community contributions, suggestions and constructive feedback on our documentation. Our documentation is hosted on the [Charmhub forum](https://discourse.charmhub.io/) to enable easy collaboration. Please use the "Help us improve this documentation" links on each documentation page to either directly change something you see that's wrong, or ask a question, or make a suggestion about a potential change via the comments section.
+
+If there's a particular area of documentation that you'd like to see that's missing, please [file a bug](https://github.com/canonical/penpot-operator/issues).
```
Comment (erinecon) [inline docs/index.md:25]: Does this link need to be updated for chrony?

File: docs/tutorial/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# Tutorial: Deploy the Chrony charm
+
+## What you'll do
+This tutorial will walk you through deploying the chrony charm; you will:
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:4]: Nitpick: Capitalise Chrony (assuming it needs to be capitalised; it seems like it based on the rest of the tutorial)

File: docs/tutorial/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+## Create a Juju model
+Since the clock is a kernel function, we will need to use LXD virtual machines
+instead of containers.
+```
+juju add-model test-chrony
+juju set-model-constraints virt-type=virtual-machine
+```
+
+## Deploy the chrony charm and confirm the chrony charm are deployed
+1. Deploy the latest chrony charm.
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:22]: Nitpick: Remove the explicit numbering of steps (reads more like a how-to guide with numbered steps)

File: docs/tutorial/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+```
+juju deploy chrony --channel latest/edge
+```
+2. Run `juju status`. You should see something like this (IPs will be different):
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:26]: Nitpick: Remove the explicit numbering of steps

File: docs/tutorial/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+Since Chrony needs a time source before it can serve time to others, 
+we must configure a time source for the Chrony charm. 
+In this example, we will use an upstream NTP server (ntp.ubuntu.com) 
+as the time source and set the iburst option to true for this source.
+
+```
+juju config chrony "sources=ntp://ntp.ubuntu.com?iburst=true"
+```
+
+Once the source is configured, the chrony charm should enter the active state.
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:51]: Nitpick: Be more explicit that the charm shows as active after running `juju status`, so something like

> Once the source is configured, the chrony charm should enter the active state in the `juju status` table.

File: docs/tutorial/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+Last offset     : +0.000024886 seconds
+RMS offset      : 0.000024886 seconds
+Frequency       : 31.247 ppm slow
+Residual freq   : +3.838 ppm
+Skew            : 67.618 ppm
+Root delay      : 0.231240883 seconds
+Root dispersion : 0.015818920 seconds
+Update interval : 2.0 seconds
+Leap status     : Normal
+```
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:147]: Add a "Clean up the environment" section to provide a satisfying end to the tutorial :)

File: docs/tutorial/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+Unit       Workload  Agent  Machine  Public address  Ports    Message
+chrony/0*  blocked   idle   0        10.81.171.230   123/udp  no time source configured
+
+Machine  State    Address        Inst id        Base          AZ  Message
+0        started  10.81.171.230  juju-728b65-0  ubuntu@24.04      Running
+```
+
+## Configure the time source for chrony charm
+Since Chrony needs a time source before it can serve time to others, 
+we must configure a time source for the Chrony charm.
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:43]: Nitpick: For conciseness, I would rephrase this sentence as "We must configure a time source for the Chrony charm so it can serve time to others."

For my understanding, the Chrony charm serves time to other...charms? Might be useful to clarify if it's a specific type of "others"

File: docs/tutorial/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+After that, enable NTS in the Chrony client configuration and restart the 
+Chrony client.
+
+```
+server ntp.example.com nts iburst certset 1
+ntstrustedcerts 1 /etc/chrony/nts.crt
+```
+
+Wait a few seconds, and you should see that the Chrony client is synchronized
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:129]: Nitpick: synchronized --> synchronised

**Review by erinecon — COMMENTED** [review]
Hi Weii, the tutorial looks great -- thanks for your work on it! 

I mostly have minor comments, with my one blocker being for you to add a "Clean up the environment" section at the end.

File: docs/index.md (hunk 1 above)
Comment (weiiwang01) [reply docs/index.md]: Oops, updated, thanks!

File: docs/tutorial/getting-started.md (hunk 2 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md:4]: Hmm, maybe we should standardize all mentions to use the lowercase "chrony"? I did some research online, it appears that the official documents and articles mostly use the lowercase version.

Some examples:
Official chrony website: [chrony-project.org](https://chrony-project.org)
Chrony FAQ: [chrony-project.org/faq.html](https://chrony-project.org/faq.html)
Article by the current chrony maintainer: https://www.redhat.com/en/blog/combining-ptp-ntp-get-best-both-worlds

File: docs/tutorial/getting-started.md (hunk 3 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Removed, thanks!

File: docs/tutorial/getting-started.md (hunk 4 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Removed, thanks!

File: docs/tutorial/getting-started.md (hunk 7 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Yes, indeed this is better, and chrony charm serves time to all other NTP clients, so how about: "We must configure a time source for the Chrony charm so it can serve time other NTP clients."?

File: docs/tutorial/getting-started.md (hunk 5 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Updated, thanks!

File: docs/tutorial/getting-started.md (hunk 8 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Updated, thanks!

File: docs/tutorial/getting-started.md (hunk 6 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Yes, I think this is a good idea. However, I’m not sure how much cleanup we can accomplish for this tutorial. The issue here  that the `chrony` package we installed via `apt` is somewhat "destructive". Installing it will replace the system’s default NTP client with `chrony`, and I’m unsure if uninstalling it will restore the original NTP client. Cleanup could involve reverting the changes made to the `chrony` configuration file on the host system back to its default settings, while leaving `chrony` installed. This should allow the reader’s system to functionally back to before, especially since on ubuntu 25.04 and above, `chrony` becomes the default NTP client anyway, so no changes will occur in that case. Alternatively, we could consider uninstalling `chrony` and fully restoring the system to its prior state, which is more complex. To simplify this, we might revise the tutorial to install the `chrony` NTP client within an LXD VM rather than directly on the host machine.

File: docs/tutorial/getting-started.md (hunk 2 above)
Comment (erinecon) [reply docs/tutorial/getting-started.md:4]: Thanks for doing the research! I'm fine with using lowercase "chrony", as long as it's consistent throughout the documentation :)

File: docs/tutorial/getting-started.md (hunk 7 above)
Comment (erinecon) [reply docs/tutorial/getting-started.md]: Thank you for the clarification! Your suggestion looks good but might be missing a "to"? More explicitly:

> "We must configure a time source for the Chrony charm so it can serve time **to** other NTP clients."

File: docs/tutorial/getting-started.md (hunk 6 above)
Comment (erinecon) [reply docs/tutorial/getting-started.md]: Thank you for your thoughts and the explanation why a cleanup section is tricky to include. I'm leaning toward installing the `chrony` NTP client within a VM, but my question to you is, how much extra work would it be for you to rework the tutorial?

File: docs/tutorial/getting-started.md (hunk 2 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md:4]: Thanks, replaced all instances of "Chrony" with "chrony"

File: docs/tutorial/getting-started.md (hunk 6 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: I have updated the document to install chrony as the NTP client in an LXD virtual machine. To differentiate which commands should run on the host machine and which should run on the virtual machine, I used the command line prompts to clarify this (`ubuntu@laptop:~$` vs `root@ntp-client:`). I have also added a section at the end for cleanup, which is very simple: just delete the virtual machine and the Juju model.

File: docs/tutorial/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
+# Tutorial: Deploy the chrony charm
+
+## What you'll do
+This tutorial will walk you through deploying the chrony charm; you will:
+1. Deploy and configure the chrony charm
+2. Enable NTS for the chrony charm
+3. Use the chrony charm as a time server
+
+## Prerequisites
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:9]: Nitpick: Prerequisites --> Requirements

File: docs/tutorial/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+
+```
+ubuntu@laptop:~$ juju config chrony "sources=ntp://ntp.ubuntu.com?iburst=true"
+```
+
+Once the source is configured, the chrony charm should enter the active state 
+in the `juju status` table.
+
+## Use the chrony Charm as a Time Source
+You can now use the chrony charm as a time source to set up time synchronization
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:55]: Nitpick: synchronization --> synchronisation

File: docs/tutorial/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+Residual freq   : +3.838 ppm
+Skew            : 67.618 ppm
+Root delay      : 0.231240883 seconds
+Root dispersion : 0.015818920 seconds
+Update interval : 2.0 seconds
+Leap status     : Normal
+```
+
+## Clean up the environment
+You can now clean up the environment by removing the juju model and the NTP
```
Comment (erinecon) [inline docs/tutorial/getting-started.md:160]: Nitpick: Capitalise Juju

File: docs/tutorial/getting-started.md (hunk 6 above)
Comment (erinecon) [reply docs/tutorial/getting-started.md]: Thanks so much for updating! Typically we're not supposed to use command line prompts (according to the style guide), but I think it's fine in this case since we need to distinguish between two different machines.

File: docs/tutorial/getting-started.md (hunk 9 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Updated, thanks!

File: docs/tutorial/getting-started.md (hunk 10 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: updated, thanks!

File: docs/tutorial/getting-started.md (hunk 11 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Updated, thanks!

File: docs/tutorial/getting-started.md (hunk 7 above)
Comment (weiiwang01) [reply docs/tutorial/getting-started.md]: Updated, thanks!

**Review by erinecon — COMMENTED** [review]
The tutorial looks great! I approve of the documentation changes :+1:
